### PR TITLE
Navigator bar menu

### PR DESCRIPTION
**Pull request by Amirk390**
+ Added navigator bar style
+ Added navigator bar functionality 
  + Routes
  + Drop down menu (display: none and display: flex)

Reference #16

### CODE REVIEW

**Review by SamiSha99 — APPROVED** [review]
Looks good for me.

**Review by matthewdking — COMMENTED** [review]
@SamiSha99 it would be good if you could have some comments or request changes! I would say unless the PR is very small **there will always** be mistakes or changes to be made? Or even questions to be asked?

@Amirk390 are you implementing the props changes onClick we were discussing today in this PR?

**Comment by Amirk390** [comment]
@matthewdking yeah
Im going to work on the props changes

**Review by matthewdking — COMMENTED** [review]
If you create a pull request but are still working on it, it's good practice to add [WIP] at the start of the PR title. This stands for **Work In Progress** and then I know not to review it because you are still working on it :construction:

**Comment by Amirk390** [comment]
@matthewdking We are good to go now :smile:

**Review by SamiSha99 — COMMENTED** [review]
Are you sure these flex attributes are affecting the code at all?

File: src/App.jsx [hunk 1]
```diff
@@ -1,10 +1,29 @@
...
+  display: flex;
+  justify-content: center;
+  flex-direction: row;
+  align-items: center;
+  width: 100%;
+  background-color: #383854;
 `;
 
-const App = () => <Title>Hey Amir and Sami and Mario</Title>;
+// our functionality
```
Comment (matthewdking) [inline src/App.jsx]: This comment and empty lines seem unnecessary as there is no code before the next comment

File: src/NavBar.jsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  flex-direction: column;
+  justify-content: center;
+`;
+
+class NavBar extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      clicked: false,
```
Comment (matthewdking) [inline src/NavBar.jsx]: clicked isn't the best naming! I think something like `menuClick` or just `menu` or `dropdownClick` or `dropdown` or anything else that is more descriptive about what you are clicking

clicked could mean the user has clicked anything? What if you add something else that gets clicked and want to change another thing in the state?

File: src/NavBar.jsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            </DropdownBtn>
+            <Title>Water You Talking About?</Title>
+          </MenuContainer>
+        </Nav>
+        <Nav2>
+          <DropdownMenu show={this.state.clicked} id="button">
+            <HLink to="/">Home</HLink>
+            <HLink to="/calculator">Calculator</HLink>
+            <HLink to="prices">Prices</HLink>
+            <HLink to="statistis">Statistics</HLink>
```
Comment (matthewdking) [inline src/NavBar.jsx]: typo statistis => statistics

Comment (matthewdking) [reply src/NavBar.jsx]: Also both the links above should have a `/` in front of them.

**Review by matthewdking — CHANGES_REQUESTED** [review]
Few minor changes

File: src/NavBar.jsx [hunk 4]
```diff
@@ -131,15 +129,15 @@ class NavBar extends Component {
...
+      menuClick: false,
     };
 
     this.handleClick = this.handleClick.bind(this);
   }
 
   handleClick() {
     this.setState({
-      clicked: !this.state.clicked,
+      menuClick: !this.state.menuClick,
```
Comment (Amirk390) [inline src/NavBar.jsx]: Nice, now it's more understandable :+1:

File: src/NavBar.jsx [hunk 5]
```diff
@@ -58,15 +58,32 @@ const DropdownBtn = styled.a`
...
+  0% {
+    height: 0rem;
+    margin: 0 0 0rem 0;
+   }
+  100% {
+    height: 1.6rem;
+    margin: 0 0 1rem 0;
+   }
+  `;
+
```
Comment (Amirk390) [inline src/NavBar.jsx:68]: Niiice :tada: 
I like the new animation 
good job Sami :clap:

**Comment by Karyum** [comment]
@Amirk390 approve PR and reassign @matthewdking so he could review this

**Comment by Amirk390** [comment]
@Karyum 
> approve PR and reassign @matthewdking so he could review this

i cant, thats my own PR

image

**Comment by Karyum** [comment]
@Amirk390 lol k then no worries

File: src/App.jsx [hunk 6]
```diff
@@ -1,10 +1,27 @@
...
+import NavBar from './NavBar';
 
-const Title = styled.h1`
-  color: blue;
+// Styling section
+
+const Container = styled.div`
+  display: flex;
+  justify-content: center;
+  flex-direction: row;
```
Comment (matthewdking) [inline src/App.jsx]: this is the default you should be able to remove this line

File: src/NavBar.jsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+  width: 100%;
+`;
+
+const Title = styled.h1`
+  display: flex;
+  justify-content: center;
+  border: 2px solid #a3bdff;
+  border-radius: 1rem;
+  color: #a3bdff;
+  font-size: 24px;
```
Comment (matthewdking) [inline src/NavBar.jsx]: font sizes should also be rem. Same below :point_down:

File: src/NavBar.jsx [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    font-size: 22px;
+    width: 11.5em;
+  }
+
+  @media only screen and (max-width: 400px) {
+    font-size: 22px;
+  }
+`;
+
+const Nav2 = styled.nav`
```
Comment (matthewdking) [inline src/NavBar.jsx]: Nav2 is not good naming please think up a more descriptive name

File: src/NavBar.jsx [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+const DropdownBtn = styled.a`
+  display: flex;
+  position: absolute;
+  left: 0;
+  align-items: flex-start;
+  color: #a3bdff;
+  margin: 0 0 0 1rem;
+`;
+
+const animnavbar = keyframes`
```
Comment (matthewdking) [inline src/NavBar.jsx]: this is JavaScript and I think camelCasing variable names is good practice here.

File: src/NavBar.jsx [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+    margin: 0 0 1rem 0;
+   }
+  `;
+
+const DropdownMenu = styled.div`
+  display: ${props => (props.show ? 'flex' : 'none')};
+  justify-content: center;
+  text-align: center;
+  background-color: #383854;
+  font-size: 18px;
```
Comment (matthewdking) [inline src/NavBar.jsx]: font-size same as above

**Review by matthewdking — CHANGES_REQUESTED** [review]
Minor changes. I think just go through everything so far and change `px` to `rem`

File: src/App.jsx (hunk 6 above)
Comment (Amirk390) [reply src/App.jsx]: Thanks, i didnt know this :smile:

**Review by SamiSha99 — APPROVED** [review]

File: src/NavBar.jsx [hunk 11]
```diff
@@ -0,0 +1,167 @@
+import React, { Component } from 'react';
+import styled, { keyframes } from 'styled-components';
+import { Link } from 'react-router-dom';
+import './styles/reset';
+
+const MenuContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  flex-direction: row;
```
Comment (matthewdking) [inline src/NavBar.jsx]: Missed this one before sorry :see_no_evil:

File: src/NavBar.jsx [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+  align-items: center;
+  width: 100%;
+`;
+
+const Nav = styled.nav`
+  display: flex;
+  flex-direction: row;
```
Comment (matthewdking) [inline src/NavBar.jsx]: ...and here

File: src/NavBar.jsx [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+
+  @media only screen and (max-width: 400px) {
+    font-size: 0.8rem;
+  }
+`;
+
+const Nav2 = styled.nav`
+  display: flex;
+  align-content: center;
+  flex-direction: row;
```
Comment (matthewdking) [inline src/NavBar.jsx]: and here

File: src/NavBar.jsx [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+    font-size: 1rem;
+    width: 11.5em;
+  }
+
+  @media only screen and (max-width: 400px) {
+    font-size: 0.8rem;
+  }
+`;
+
+const Nav2 = styled.nav`
```
Comment (matthewdking) [inline src/NavBar.jsx]: This is still the same name?

**Review by matthewdking — CHANGES_REQUESTED** [review]

**Review by matthewdking — APPROVED** [review]
Noice :whale:

**Pull request merged**